src/extract.jl

```diff
@@ -372,7 +372,7 @@ function read_tarball(
             hdr′ = Header(hdr′, link=link)
             what = get(paths, link, Symbol("non-existent"))
             if what isa Integer # plain file
-                hdr′ = Header(hdr′, size=what)
+                hdr′ = Header(hdr′, size=Int64(what)::Int64)
```

Comment:
Honestly, I'm always saddened when I see stuff like `Int64(x)::Int64` to improve inference :pensive:

Reply:
Here it's `Int64` rather than `Int` because that's what the `Header` struct uses. But the only thing here that Julia knows is that `what isa Integer`, and there is no upper limit on the number of methods that `Int64(what)` might theoretically call, depending on how many `Integer` subtypes have been defined.

Reply:
You should be happier now :smile:

Reply:
🥳